pkg/meta/model/job.go

```diff
@@ -512,11 +513,15 @@ func (job *Job) Encode(updateRawArgs bool) ([]byte, error) {
 				}
 			}
 		} else {
-			job.RawArgsV2, err = json.Marshal(job.ArgsV2)
+			var arg any
+			if len(job.Args) > 0 {
+				arg = job.Args[0]
+			}
+			job.RawArgs, err = json.Marshal(arg)
```

Comment:
if can moved inside `if len(job.Args) > 0`? And do you think we should assert `len(job.Args) == 1`

Reply:
no, we need Marshal nil to clear args out

> And do you think we should assert len(job.Args) == 1

add in another PR, https://github.com/pingcap/tidb/pull/55964/files#diff-c51ad6b394f865592721a3e18d58574176394570e3efcff7703e8bd0c3c5c2efR518